core/controllers/certificate_assessment.py

```
@@ -450,24 +450,51 @@ class SubmitCertificateAssessmentHandler(
+        self.render_json(
+            {'attempt_id': attempt.attempt_id, 'is_submitted': True}
+        )
+
+
+class CertificateQuestionHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Fetches question state data for an in-progress certificate attempt."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {'question_id': {'schema': {'type': 'basestring'}}}
```

Ruff (0.16.0)

[warning] Mutable default value for class attribute

(RUF012)